src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -146,23 +177,19 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, gdDAO: Datap
     }
   }
 
-  def getClusterDetails(googleProject: GoogleProject, clusterName: String): Future[Cluster] = {
-    dbRef.inTransaction { dataAccess =>
-      getCluster(googleProject, clusterName, dataAccess)
-    }
-  }
-
-  def deleteCluster(googleProject: GoogleProject, clusterName: String): Future[Int] = {
-    getClusterDetails(googleProject, clusterName) flatMap { cluster =>
-      if(cluster.status.isActive) {
-        for {
-          _ <- gdDAO.deleteCluster(googleProject, clusterName)
-          recordCount <- dbRef.inTransaction(dataAccess => dataAccess.clusterQuery.markPendingDeletion(cluster.googleId))
-        } yield {
-          clusterMonitorSupervisor ! ClusterDeleted(cluster)
-          recordCount
+  private[service] def processLabelMap(labelMap: Map[String, String]): Map[String, String] = {
+    // Explode the parameter '_labels=key1=value1,key2=value2' into a Map of keys to values.
+    // This is to support swagger which doesn't allow free-form query string parameters.
+    val extraLabelMap = labelMap.get("_labels") match {
+      case Some(extraLabels) if extraLabels.nonEmpty =>
+        extraLabels.split(',').foldLeft(Map.empty[String, String]) { (r, c) =>
+          c.split('=') match {
+            case Array(key, value) => r + (key -> value)
+            case _ => throw ParseLabelsException(extraLabels)
+          }
         }
-      } else Future.successful(0)
+      case _ => Map.empty
     }
+    (labelMap - "_labels") ++ extraLabelMap
```

Comment:
I'm confused - why is the returned object not just `extraLabelMap`? Why are taking out `_labels` from labelMap? What else does labelMap contain aside from `_labels`?

Reply:
I did it in case the styles are mixed:
```
http://leo/api/clusters?foo=bar&_labels=baz%3Dbiz
// would generate: Map(foo -> bar, baz -> biz)
```
Not that I would ever expect that to happen. Maybe if `_labels` is specified we should just use that..

Reply:
Ohhhh, I see

Reply:
Yeah, I think we don't have to account for the styles to be mixed so do either/or instead?

But whether or not you account for mixed style, maybe add a comment to the code to explain like you did up in the api-docs.yaml file?

Reply:
👍